### PR TITLE
Make bandoracled support execution to aws lambda

### LINKED ISSUE #830: Make bandoracled support execution to AWS lambda
Ask @sorawit how to execute things on AWS lambda.

### PR DESCRIPTION
**Pull request by prin-r**
Fix #830

### CODE REVIEW

File: chain/byteexec/byteexec.go [hunk 1]
```diff
@@ -90,3 +93,48 @@ func RunOnDocker(executable []byte, timeOut time.Duration, arg string) ([]byte,
 
 	return exec.CommandContext(ctx, "docker", newArgs...).Output()
 }
+
+// RunOnAWSLambda runs the given executable on AMS Lambda platform.
+func RunOnAWSLambda(executable []byte, timeOut time.Duration, arg string, url string) ([]byte, error) {
+	requestBody, err := json.Marshal(map[string]string{
```
Comment (taobun) [inline chain/byteexec/byteexec.go]: Use `grequests` library to wrap the request.

File: chain/cmd/bandoracled/main.go [hunk 2]
```diff
@@ -23,6 +23,7 @@ import (
 const (
 	flagMaxQueryDuration = "max-query-duration"
 	flagPrivKey          = "priv-key"
+	flagReporterURL      = "reporter-url"
```
Comment (taobun) [inline chain/cmd/bandoracled/main.go]: flagExecuteEndPoint = "execute-endpoint"

File: chain/cmd/bandoracled/main.go [hunk 3]
```diff
@@ -119,6 +120,14 @@ $ bandoracled --node tcp://localhost:26657 --priv-key 06be35b56b048c5a6810a47e2e
 		"Private key of validator to send report transaction",
 	)
 	viper.BindPFlag(flagPrivKey, cmd.Flags().Lookup(flagPrivKey))
+
+	cmd.Flags().String(
+		flagReporterURL,
+		"",
+		"The URL of reporter end-point which will receive 3 parameters (executable, timeout, calldata)",
```
Comment (taobun) [inline chain/cmd/bandoracled/main.go]: execute

**Review by taobun — CHANGES_REQUESTED** [review]

File: chain/cmd/bandoracled/main.go [hunk 4]
```diff
@@ -173,12 +182,25 @@ func handleRequest(requestID zoracle.RequestID) {
...
-			result, err := byteexec.RunOnDocker(
-				dataSource.Executable,
-				viper.IsSet(flagSandboxMode),
-				time.Duration(viper.GetInt(flagMaxQueryDuration))*time.Second,
-				string(calldata),
-			)
+			executeEndPoint := viper.GetString(flagExecuteEndPoint)
+
+			var result []byte
+			if executeEndPoint == "" {
```
Comment (taobun) [inline chain/cmd/bandoracled/main.go]: Use viper.IsSet(flagExecuteEndPoint)

**Review by taobun — CHANGES_REQUESTED** [review]

**Review by taobun — APPROVED** [review]

**Review by sorawit — APPROVED** [review]

**Pull request merged**